From 2c2cbb176658deade36897bb0e1eaecb695e4e98 Mon Sep 17 00:00:00 2001
From: Andrei-Liviu Simion <andrei.simion@xilinx.com>
Date: Fri, 1 Apr 2016 15:13:31 -0700
Subject: [PATCH 0967/1566] drm: xilinx: dp: Maximum pre-emphasis of 2 for
 ZynqMP.

commit  16d762973ab7052592a57daa045165e79bc9826e from
https://github.com/Xilinx/linux-xlnx.git

DisplayPort on ZynqMP has a maximum pre-emphasis level of 2.
DisplayPort in soft IP has a maximum level of 3.

Signed-off-by: Andrei-Liviu Simion <andrei.simion@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_dp.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
index 90103d1..279e33f 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
@@ -454,6 +454,7 @@ static void xilinx_drm_dp_adjust_train(struct xilinx_drm_dp *dp,
 {
 	u8 *train_set = dp->train_set;
 	u8 voltage = 0, preemphasis = 0;
+	u8 max_preemphasis;
 	u8 i;
 
 	for (i = 0; i < dp->mode.lane_cnt; i++) {
@@ -470,7 +471,10 @@ static void xilinx_drm_dp_adjust_train(struct xilinx_drm_dp *dp,
 	if (voltage >= DP_TRAIN_VOLTAGE_SWING_LEVEL_3)
 		voltage |= DP_TRAIN_MAX_SWING_REACHED;
 
-	if (preemphasis >= DP_TRAIN_PRE_EMPH_LEVEL_3)
+	max_preemphasis = (dp->dp_sub) ? DP_TRAIN_PRE_EMPH_LEVEL_2 :
+					 DP_TRAIN_PRE_EMPH_LEVEL_3;
+
+	if (preemphasis >= max_preemphasis)
 		preemphasis |= DP_TRAIN_MAX_PRE_EMPHASIS_REACHED;
 
 	for (i = 0; i < dp->mode.lane_cnt; i++)
-- 
1.7.5.4

